sdk/servicebus/service-bus/src/core/linkEntity.ts

```diff
@@ -235,8 +235,7 @@ export abstract class LinkEntity<LinkT extends Receiver | AwaitableSender | Requ
     }
 
     if (this._wasClosedPermanently) {
-      log.error(`${this._logPrefix} Link has been closed. Not reopening.`);
-      return;
+      throw new Error(`${this._logPrefix} Link has been permanently closed. Not reopening.`);
```

Comment:
Is there any scenario where the user would see this? The user facing operations have this check on whether sender/receiver/client is closed and throw error appropriately. When do we see a real user encountering this error?

Reply:
This is more just being defensive in this code although (when we failed to properly do the checks external to this) I did run into this.

I think it's a solid thing to have in here since it fits with the contract of the method - if we do not create a link in this method we throw.

Reply:
(well, not create a link meaning "failed to exit this method with an open link")